### ISSUE TITLE
Support telescope

### ISSUE THREAD

**Issue by SingularisArt** [gennaro-tedesco/nvim-possession#2]
Hello! I really want to start using sessions in neovim, but I've always seen the session plugins to be bloated with a bunch of features that I don't need, which is why I'm currently using yours.
Do you think you could support `telescope.nvim`? Like, in the setup, you could specify what you want to use, like:

```lua
require("nvim-possession").setup({
  viewer = "telescope|fzf",
  ...
})
```

**Comment by gennaro-tedesco** [comment]
> Do you think you could support telescope.nvim? Like, in the setup, you could specify what you want to use, like:

yes, this is planned for future releases. However, I am not able to estimate when exactly simply because - not using Telescope myself - I am not familiar with how they implement extensions and I would need some time to look into it and learn it. I will keep this issue open since other people asked for it already as well, and they can refer to this.

**Comment by SingularisArt** [comment]
I'll help you with that over the weekend, since I use telescope for pretty much everything.

**Comment by gennaro-tedesco** [comment]
That's great! Meanwhile I will start looking into it as well :)

**Comment by tmpm697** [comment]
+1

**Comment by SingularisArt** [comment]
Alright. I've got the basics working. I'll create a PR tomorrow with a more refined way of doing things.

**Comment by tmpm697** [comment]
@SingularisArt can you decouple it to a separate plugin instead of an PR to nvim-possession?

**Comment by gennaro-tedesco** [comment]
> Alright. I've got the basics working. I'll create a PR tomorrow with a more refined way of doing things.

thank you very much @SingularisArt for your help, I am looking forward to the PR! Notice I _might_ be a little slow in responding/reviewing the code within the week as I am at work (can only dedicate some time in the evenings) but I do see the notifications and all the rest, so please bear a little patience :)

> can you decouple it to a separate plugin instead of an PR to nvim-possession?

I still think it can be of use to include it here unless the codebase for a Telescope extension is huge and changes too much.

**Comment by SingularisArt** [comment]
unfortunately, it does look like a huge chunk of the code will be rewritten. I think it'll be best if we put this in a separate repo, that's specifically for this telescope extension because we have to create the folder `lua/telescope/_extension/*`.
IDK. What are your thoughts?

**Comment by gennaro-tedesco** [comment]
Hmm, I see, this is what I feared too, if I am being honest. What part of the code must be re-written?

I am asking it because the original idea of this plugin is not to be a replacement of the other fully fledged plugins (especially Telescope) but rather to be a light extension that doesn't need much configuration or thinking. So whilst we are adding new features (see [v0.0.2](https://github.com/gennaro-tedesco/nvim-possession/releases/tag/v0.0.2) for instance) and Telescope support would be awesome, I wouldn't want it to come at the price of complete refactoring/restructuring of the codebase.

**Comment by SingularisArt** [comment]
I've created a PR.
